### ISSUE TITLE
Getting some lines in the transcription repeating multiple times

### ISSUE THREAD

**Issue by iamianM** [sanchit-gandhi/whisper-jax#88]
I'm consisting getting some lines back that repeat multiple times in the transcript like so:

`[43:51.000 -> 44:48.720]  You guys know the general conceit of AlphaGo, which is that you have the AI play itself in Go 44 million times in a couple of hours, and so doing it becomes better than any known human player. It turns out a lot of AI is now based on this kind of self-play idea. Well, here's a new game. You're given a secret topic, I'm given a secret topic. I'm trained to get you to say positive things about my topic, you're doing the same. Whoever gets the other person to do it most wins. Well, to do that, I have to model what you're trying to get me to say and I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it. And I have to be able to do it same. Whoever gets the other person to do it most, wins. Well, to do that, I have to model what you're trying to get me to say, and I have to figure out how to persuade you to say what I want to say. This is not alpha go, this is alpha persuade. And this is completely possible with today's technology, and in so doing, it'll become better than any known human at persuasion.`

Any idea why this happens and how to stop it. 

For context, I am using a t2-8 instance with the medium model.

**Comment by HarutyunyanLiana** [comment]
Whisper has the problem of hallucinations, and this is one example of it (maybe you have some silences in the audio?). I think transcribing without any context makes it hallucinates more. One way that I find useful, if you transcribe with timestamps, you will have segments, and you can check if the segment is the same as the previous one, drop it.

**Comment by Brodski** [comment]
I haven't had too much success with whisper-jax, but [faster-whisper](https://github.com/guillaumekln/faster-whisper) is workings better for me. Both for performance and for the issue you mentioned OP. I know this isnt the solution you're looking for, but I'm using faster-whisper instead, and this does it for me:

`segments, info = model.transcribe(audio_path, language="en", condition_on_previous_text=False, vad_filter=True)`

**Comment by AlexandderGorodetski** [comment]
Hello @HarutyunyanLiana . Do you mean that segments are exactly same in terms of start_time and end_time and "text" or do you mean "text" only?

**Comment by AlexandderGorodetski** [comment]
Guys, this is weird. I found that this hallucination problem exists for small and medium models, but it does not exist for tiny model. Seems that there is some problem not in decoding algorithm but there is some issue with the training of small and medium models.

**Comment by Brodski** [comment]
you might come across the problem on different audio files. IIRC it's a problem dues to silence or music (non words). It's a issue reported in other repos (faster-whisper issues is where I read it)

**Comment by projects-g** [comment]
I installed the latest available whisper codebase but still got the same error. Some particular sets of words keep repeating over and over. Has the issue resurfaced?


 `And the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, the number two thing, holding these `

**Comment by patriotyk** [comment]
I don't think it is whisper model problem, I have file that works without any issues on other implementations. I can share it if you want.
This is how transcript looks: 
```[00:00.000 -> 00:03.500]  Sweet dreams are made of this
[00:03.500 -> 00:07.000]  Who am I to disagree?
[00:07.000 -> 00:11.000]  I travel the world and the seven seas
[00:11.000 -> 00:15.000]  Everybody's looking for something
[00:15.000 -> 00:18.000]  Some of them want to use you
[00:19.000 -> 00:22.500]  Some of them want to get used by you
[00:22.500 -> 00:26.000]  Some of them want to get used by you Some of them want to abuse you Some of them want to be abused```

**Comment by projects-g** [comment]
I still suspect Whisper, since when there is silence, the model puts out sentences similar to the ones I posted above.

**Comment by Zeiny96** [comment]
Try using condition_on_previous_text=False, hallucination is a common issue in whisper when condition_on_previous_text is set to True especially with silences or music in the audio
